ReactNativeClient/lib/services/RecordAudio.js

```diff
@@ -0,0 +1,96 @@
+import AudioRecorderPlayer from 'react-native-audio-recorder-player';
+import { PermissionsAndroid, Alert, Platform } from 'react-native';
+
+class RecordAudio {
+	constructor() {
+		this.dispatch = () => {};
+		this.audioRecorderPlayer = new AudioRecorderPlayer();
+		this.recordSecs = 0;
+		this.recordTime = 0;
+	}
+
+	static instance() {
+		if (RecordAudio.instance_) return RecordAudio.instance_;
+		RecordAudio.instance_ = new RecordAudio();
+		return RecordAudio.instance_;
+	}
+
+	async onStartRecord() {
+		if (Platform.OS === 'android') {
+			try {
+				const granted = await PermissionsAndroid.request(
```

Comment:
PermissionsAndroid might not work correctly: https://github.com/laurent22/joplin/blob/142976f012d9398bfd2e715bd6fb4bf32dc7b5d5/ReactNativeClient/lib/components/screens/config.js#L468

And indeed on my phone it always returns `never_ask_again`

react-native-permissions is a good alternative.

Reply:
Ignore that, tested again and PermissionsAndroid works now.